library/src/main/java/com/pokegoapi/api/map/fort/Raid.java

```diff
@@ -0,0 +1,112 @@
+/*
+ *     This program is free software: you can redistribute it and/or modify
+ *     it under the terms of the GNU General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     (at your option) any later version.
+ *
+ *     This program is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ *
+ *     You should have received a copy of the GNU General Public License
+ *     along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.pokegoapi.api.map.fort;
+
+import POGOProtos.Data.Raid.RaidInfoOuterClass.RaidInfo;
+import POGOProtos.Map.Fort.FortDataOuterClass;
+import com.pokegoapi.api.PokemonGo;
+import com.pokegoapi.google.common.geometry.S2LatLng;
+import lombok.Getter;
+
+public class Raid {
+
+	private final PokemonGo api;
+	@Getter
+	private final FortDataOuterClass.FortData fortData;
+	@Getter
+	private long cooldownCompleteTimestampMs;
+	@Getter
+	private final RaidInfo raidInfo;
+
+	
+	public Raid(PokemonGo api, FortDataOuterClass.FortData fortData) {
+		this.api = api;
+		this.fortData = fortData;
+		this.raidInfo = fortData.getRaidInfo();
+	}
+
+	
+	public double getDistance() {
+		S2LatLng pokestop = S2LatLng.fromDegrees(getLatitude(), getLongitude());
+		S2LatLng player = S2LatLng.fromDegrees(api.getLatitude(), api.getLongitude());
+		return pokestop.getEarthDistance(player);
+	}
+	
+	public long getRaidSeed(){
+    	return raidInfo.getRaidSeed();
+    }
+    public long getRaidSpawnMs(){
+    	return raidInfo.getRaidSpawnMs();
+    }
+
+    public long getRaidBattleMs(){
+    	return raidInfo.getRaidBattleMs();
+    }
+
+    public long getRaidEndMs(){
+    	return raidInfo.getRaidEndMs();
+    }
+
+    public boolean hasRaidPokemon(){
+    	return raidInfo.hasRaidPokemon();
+    }
+    
+    public POGOProtos.Data.PokemonDataOuterClass.PokemonData getRaidPokemon(){
```

Comment:
This should be wrapped in the `PokemonData` object provided by the API, and initialized in the constructor. Ideally, the proto should not be stored in memory.